lib/orderbook/OrderBook.ts

```diff
@@ -279,19 +268,19 @@ class OrderBook extends EventEmitter {
         this.emit('ownOrder.filled', portion);
         onUpdate && onUpdate({ case: PlaceOrderEventCase.InternalMatch, payload: maker });
       } else {
-        if (!this.swaps || !this.swaps.verifyExecution(maker, taker)) {
-          rejectNonInternalMatch(taker, maker);
+        if (!this.swaps) {
           continue;
```

Comment:
You are missing here the part of appending the taker order to the remaining order. It will cause a portion of the taker order to disappear.

Reply:
We're already doing it when handling the swap failures. Instead, we can do it here and also after an exception is thrown (line 280), so it will be more clear that it's happening on both of these cases.

Reply:
OK, I was mentally treating the case of no swaps module differently because that should only happen during testing, but I'll treat it the same as a swap failure for now.

Reply:
This is not optimal IMO, because you will have a few futile swap retries iterations untill reaching the timeout, where none can actually succeed.
But since this condition is solely for testing, i'm fine with not over-optimizing it, also because this whole area is subject to future changes anyway.